src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala

```diff
@@ -2,59 +2,77 @@ package org.zalando.spearheads.innkeeper.api
 
 import java.time.LocalDateTime
 
-import org.zalando.spearheads.innkeeper.api.Endpoint.{ EndpointType, Https, Protocol, ReverseProxy }
-import org.zalando.spearheads.innkeeper.api.PathMatcher.MatcherType
-
 import scala.collection.immutable.Seq
 
 /**
  * @author dpersa
  */
-case class Route(id: Long, route: NewRoute, createdAt: LocalDateTime,
-                 deletedAt: Option[LocalDateTime] = None)
-
-case class Endpoint(hostname: String, path: Option[String] = None,
-                    port: Option[Int] = Some(443),
-                    protocol: Option[Protocol] = Some(Https),
-                    endpointType: Option[EndpointType] = Some(ReverseProxy))
 
-object Endpoint {
+case class RouteName(name: String) {
+  "[A-Z][A-Z0-9_]*".r.unapplySeq(name) match {
+    case None => throw InvalidRouteNameException
+    case _    =>
+  }
+}
 
-  sealed trait EndpointType
+object InvalidRouteNameException extends RuntimeException("Invalid route name. The name should match [A-Z][A-Z0-9_]*")
```

Comment:
Why not name the string conataining the regexp and use a reference here instead of duplicating it?

```scala
val validRoutePattern = "[A-Z][A-Z0-9_]*"
//...
validRoutePattern.r.unapplySeq(name) match {
//...
}
//...
object InvalidRouteNameException extends RuntimeException(s"Invalid route name. The name should match ${validRoutePattern}"
```